decidim-bulletin_board-app/app/commands/process_tally_step.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# A command with all the business logic to perform a step of the tally
+class ProcessTallyStep < Rectify::Command
+  include LogEntryCommand
+
+  # Public: Initializes the command.
+  #
+  # client - The authority or trustee sender of the tally message
+  # message_id - The message identifier
+  # signed_data - The signed message received
+  def initialize(client, message_id, signed_data)
+    @client = client
+    @message_id = message_id
+    @signed_data = signed_data
+  end
+
+  # Executes the command. Broadcasts these events:
+  #
+  # - :processed when everything is valid.
+  # - :invalid if the received data wasn't valid.
+  #
+  # Returns nothing.
+  def call
+    return broadcast(:invalid, error) unless
+      valid_log_entry?("tally")
+
+    election.with_lock do
+      return broadcast(:invalid, error) unless valid?
+
+      log_entry.election = election
+      log_entry.save!
+      create_response_log_entry!
+      election.save!
+    end
+
+    broadcast(:ok)
+  end
+
+  private
+
+  def valid?
+    valid_client?(client.authority? && election.authority == client || client.trustee? && election.trustees.member?(client)) &&
+      valid_author?(message_identifier.from_authority? || message_identifier.from_trustee?) &&
+      valid_step?(election.tally?) &&
+      process_message
+  end
```

Comment:
I think this validations are becoming more and more complex. Maybe it's the time to start some kind of `Policy` classes or something like that? We can discuss this... someday, not a priority at the moment 👍

Reply:
I can think a way to improve its readability very easily. After we merge the open PRs I can add it on a new PR